src/controllers/mainController.ts

```diff
@@ -443,7 +443,11 @@ export default class MainController implements vscode.Disposable {
 
 		// Generate Azure Function command
 		this._context.subscriptions.push(vscode.commands.registerCommand(Constants.cmdCreateAzureFunction, async (node: TreeNodeInfo) => {
-			const connectionDetails = ConnectionCredentials.createConnectionDetails(node.connectionInfo);
+			let connectionInfo = node.connectionInfo;
+			// set the database containing the selected table so it can be used
+			// for the initial catalog property of the connection string
+			connectionInfo.database = node.parentNode.parentNode.metadata.name;
```

Comment:
You should also be walking up the tree until you find a node with a nodeType === 'Database' and then grabing the name from that node. Tables aren't always guaranteed to be 2 levels below the database node, e.g. for system tables : 

![image](https://user-images.githubusercontent.com/28519865/156433910-1e976c1f-970b-4f36-877e-07dc1643e534.png)